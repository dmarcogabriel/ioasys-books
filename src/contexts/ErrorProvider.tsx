import React, {createContext} from 'react';
import {UNAUTHORIZED_STATUS} from '../constants';
import Toast from 'react-native-toast-message';

type ErrorData = {
  message: string;
};

export interface ApiError {
  errors: ErrorData;
}

interface ErrorContextValue {
  handleError: (
    error: ApiError,
    status: number,
    logoutCallback: () => Promise<void>,
  ) => Promise<void>;
}

export const ErrorContext = createContext<ErrorContextValue>({
  handleError: async () => {},
});

interface ErrorProviderProps {
  children: React.ReactNode;
}

export const ErrorProvider = ({children}: ErrorProviderProps): JSX.Element => {
  const handleError = async (
    error: ApiError,
    status: number,
    logoutCallback: () => Promise<void>,
  ): Promise<void> => {
    let errorMessage: string = '';
    if (status === UNAUTHORIZED_STATUS) {
      await logoutCallback();
      errorMessage = 'Sess√£o expirada';
    } else {
      Toast.show({
        position: 'bottom',
        text1: errorMessage || error.errors.message,
        type: 'error',
      });
    }
  };

  return (
    <ErrorContext.Provider value={{handleError}}>
      {children}
    </ErrorContext.Provider>
  );
};
