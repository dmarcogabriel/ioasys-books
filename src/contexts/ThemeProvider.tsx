import React from 'react';
import {ThemeProvider as SCThemeProvider} from 'styled-components/native';

interface ThemeProviderProps {
  children: React.ReactNode;
}

export const ThemeProvider = ({children}: ThemeProviderProps): JSX.Element => {
  const theme = {
    colors: {
      primary: '#B22E6F',
      dark: '#333333',
      light: '#ffffff',
      grey: '#999999',
    },
  };

  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};
