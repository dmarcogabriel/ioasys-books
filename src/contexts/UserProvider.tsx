import React, {createContext, useState, useEffect, useCallback} from 'react';
import {User} from '../@types/user.interface';
import {storeData, removeData, getData} from '../utils/storage';
import {AUTH_TOKEN_STORAGE_KEY} from '../constants';

interface UserContextValue {
  user: User | null;
  login: (user: User, token: string) => Promise<void>;
  logout: () => Promise<void>;
  token: string | null;
}

export const UserContext = createContext<UserContextValue>({
  user: null,
  token: null,
  login: async () => {},
  logout: async () => {},
});

interface UserProviderProps {
  children: React.ReactNode;
}

export const UserProvider = ({children}: UserProviderProps): JSX.Element => {
  const [user, setUser] = useState<User | null>(null);
  const [token, setToken] = useState<string | null>(null);
  const [isLoading, setIsLoading] = useState(true);

  const login = async (data: User, authToken: string): Promise<void> => {
    setUser(data);
    setToken(authToken);
    await storeData(AUTH_TOKEN_STORAGE_KEY, authToken);
  };

  const logout = async () => {
    await removeData(AUTH_TOKEN_STORAGE_KEY);
    setUser(null);
    setToken(null);
  };

  const loadToken = useCallback(async () => {
    setIsLoading(true);
    try {
      const authToken = await getData<string>(AUTH_TOKEN_STORAGE_KEY);
      if (authToken) {
        setToken(authToken);
      }
    } catch (error) {
      setIsLoading(false);
    }
    setIsLoading(false);
  }, []);

  useEffect(() => {
    loadToken();
  }, [loadToken]);

  return (
    <UserContext.Provider value={{user, token, login, logout}}>
      {!isLoading && children}
    </UserContext.Provider>
  );
};
