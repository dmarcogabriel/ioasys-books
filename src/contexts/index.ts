export * from './UserProvider';
export * from './ThemeProvider';
export * from './ErrorProvider';
