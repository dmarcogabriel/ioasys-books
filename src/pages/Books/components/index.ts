export * from './BookCard';
export * from './SearchInput';
export * from './BooksHeader';
export * from './BooksFooter';
