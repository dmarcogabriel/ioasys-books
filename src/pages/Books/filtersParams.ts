export const CATEGORIES = [
  'biographies',
  'collections',
  'behavior',
  'tales',
  'literary-criticism',
  'scienceFiction',
  'folklore',
  'genealogy',
  'humor',
  'children',
  'games',
  'newspapers',
  'brazilian-literature',
  'foreign-literature',
  'rare-books',
  'manuscripts',
  'poetry',
  'another-subjects',
];
