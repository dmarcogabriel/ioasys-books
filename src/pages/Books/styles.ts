import styled from 'styled-components/native';

export const SafeAreaView = styled.SafeAreaView`
  flex: 1;
`;

export const BooksContainer = styled.View`
  background: #e5e5e5;
  flex: 1;
`;

export const SearchSection = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  padding: 0 16px 16px;
  margin-top: 32px;
`;
