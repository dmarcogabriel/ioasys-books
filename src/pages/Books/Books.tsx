import React, {useCallback, useEffect, useState} from 'react';
import {TouchableOpacity, FlatList, StatusBar} from 'react-native';
import MIcon from 'react-native-vector-icons/MaterialIcons';
import {FiltersModal} from '../../components';
import {SafeAreaView, BooksContainer, SearchSection} from './styles';
import {BookCard, SearchInput, BooksHeader, BooksFooter} from './components';
import {useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import {Book} from '../../@types/book.interface';
import api from '../../services/api';
import {RootStacksParamsList} from '../../@types/rootStacksParamsList.interface';
import {useUser, useErrorHandler} from '../../hooks';

export interface ApiResponse {
  data: Book[];
  page: number;
  totalPages: number;
  totalItems: number;
}

type BooksScreenProp = NativeStackNavigationProp<
  RootStacksParamsList,
  'BookDetails'
>;

export const Books = (): JSX.Element => {
  const [books, setBooks] = useState<Book[]>([]);
  const [isLoading, setIsLoading] = useState(false);
  const [page, setPage] = useState(1);
  const [total, setTotal] = useState<number>(1);
  const [showFilters, setShowFilters] = useState<boolean>(false);
  const {navigate} = useNavigation<BooksScreenProp>();
  const {logout} = useUser();
  const {handleError} = useErrorHandler();

  const handleShowFilters = () => {
    setShowFilters(true);
  };

  const handleCloseFilters = (data: any | null) => {
    if (data) {
      console.log(data);
    }
    setShowFilters(false);
  };

  const handleOpenBookDetails = (book: Book) => {
    navigate('BookDetails', {id: book.id});
  };

  const loadBooks = useCallback(
    async (pageNumber: number) => {
      if (!isLoading) {
        setIsLoading(true);
        try {
          const {data} = await api.get<ApiResponse>('books', {
            params: {page: pageNumber},
          });
          setBooks(oldBooks => [...oldBooks, ...data.data]);
          setPage(oldPage => oldPage + 1);
          setTotal(data.totalPages);
          setIsLoading(false);
        } catch (err: any) {
          if (err.response && err.response.status) {
            handleError(err.response.data, err.response.status, logout);
          } else {
            setIsLoading(false);
          }
        }
      }
    },
    [isLoading, logout, handleError],
  );

  const loadMoreBooks = async () => {
    if (books.length < total) {
      await loadBooks(page);
    }
  };

  useEffect(() => {
    loadBooks(page);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <SafeAreaView>
      <StatusBar backgroundColor="#e5e5e5" />
      <BooksHeader />
      <BooksContainer>
        <SearchSection>
          <SearchInput onSearch={term => console.log(term)} />
          <TouchableOpacity onPress={handleShowFilters}>
            <MIcon name="tune" size={24} />
          </TouchableOpacity>
        </SearchSection>
        <FlatList
          testID="bookList"
          data={books}
          keyExtractor={item => item.id}
          renderItem={({item: book}) => (
            <BookCard
              testID={`bookCard${book.id}`}
              book={book}
              onPress={handleOpenBookDetails}
            />
          )}
          ListFooterComponent={
            <BooksFooter disabled={isLoading} onLoadMore={loadMoreBooks} />
          }
        />
        <FiltersModal isVisible={showFilters} onClose={handleCloseFilters} />
      </BooksContainer>
    </SafeAreaView>
  );
};
