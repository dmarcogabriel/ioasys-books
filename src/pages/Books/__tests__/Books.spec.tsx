import React from 'react';
import {render} from '@testing-library/react-native';
import {Books, ApiResponse} from '../Books';
import {ThemeProvider} from '../../../contexts';
import renderer from 'react-test-renderer';
import {Book} from '../../../@types/book.interface';

const mockBooks: Book[] = [
  {
    authors: ['Yango Moreira Filho', 'Norberto Santos'],
    title: 'A dolorem itaque',
    description:
      'Commodi accusamus odio quia quia sequi laboriosam. Sequi fuga est consequatur maiores maiores blanditiis necessitatibus quia animi. Ut qui officiis. Atque aut a quia corrupti neque sint est quo consectetur.\n \rDoloremque sit in qui optio qui est. Temporibus reiciendis officiis totam ut nobis itaque odit. Maiores accusantium beatae omnis natus. Officia vitae provident animi totam incidunt magni nemo. Asperiores quod nihil quia itaque culpa natus cumque quidem.',
    pageCount: 1042,
    category: 'Literatura Estrangeira',
    imageUrl: 'https://files-books.ioasys.com.br/Book-9.jpg',
    language: 'Português',
    isbn10: '3318368196',
    isbn13: '426-3318368196',
    publisher: 'Pereira - Carvalho',
    published: 1998,
    id: '1',
  },
  {
    authors: ['Ladislau Carvalho Neto', 'Rafaela Silva Jr.'],
    title: 'A voluptate a',
    description:
      'Voluptates tenetur sit quo. Dignissimos nulla nulla. Saepe veniam quas et ab eaque. Reiciendis minima ea sunt ut quae.\n \rError cum rem ducimus. Eum iste modi. Impedit ipsam eum optio. Esse velit in ut ullam natus quae nesciunt omnis a. Asperiores facere explicabo.',
    pageCount: 1342,
    category: 'Ficção Científica',
    imageUrl: 'https://files-books.ioasys.com.br/Book-0.jpg',
    language: 'Inglês',
    isbn10: '5434807270',
    isbn13: '869-5434807270',
    publisher: 'Carvalho - Nogueira',
    published: 2006,
    id: '2',
  },
  {
    authors: ['Sr. Marcela Reis', 'Natália Macedo', 'Ladislau Costa'],
    title: 'Ab',
    description:
      'Quasi eaque tempore. Rerum voluptas rem quibusdam expedita numquam ut similique. Dolorum voluptas quidem sapiente atque voluptas nihil est ipsa. Quia id explicabo esse.\n \rLaborum sit qui. Distinctio labore qui omnis odio dignissimos aut nobis. Officia expedita aspernatur totam esse. Sit voluptatem quis hic. Sapiente tempora libero. Velit et porro non mollitia.',
    pageCount: 924,
    category: 'Manuscritos',
    imageUrl: 'https://files-books.ioasys.com.br/Book-1.jpg',
    language: 'Português',
    isbn10: '6431554229',
    isbn13: '771-6431554229',
    publisher: 'Barros - Santos',
    published: 2013,
    id: '3',
  },
];
jest.mock('../../../services/api', () => ({
  get: jest.fn(
    (): ApiResponse => ({
      data: mockBooks,
      page: 1,
      totalItems: 10,
      totalPages: 1,
    }),
  ),
}));

const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: mockNavigate,
  }),
}));

const ComponentWrapper = () => (
  <ThemeProvider>
    <Books />
  </ThemeProvider>
);

describe('pages/Books', () => {
  it('should pass on render book list', async () => {
    const {getByTestId} = render(<ComponentWrapper />);

    const bookList = getByTestId('bookList');

    expect(bookList).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(renderer.create(<ComponentWrapper />).toJSON()).toMatchSnapshot();
  });
});
