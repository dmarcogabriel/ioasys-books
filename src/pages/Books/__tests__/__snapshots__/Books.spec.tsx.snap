// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Books should match snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#e5e5e5",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 40,
        },
      ]
    }
  >
    <View
      noMargin={true}
      noPadding={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginBottom": 0,
            "paddingTop": 0,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/components/HeaderTitle/assets/logo-dark.png",
          }
        }
        style={
          Array [
            Object {},
          ]
        }
      />
      <Text
        color="dark"
        style={
          Array [
            Object {
              "color": "#333333",
              "fontSize": 28,
              "marginLeft": 16,
            },
          ]
        }
        testID="headerTitle"
      >
        Books
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "#33333333",
          "borderRadius": 100,
          "borderStyle": "solid",
          "borderWidth": 1,
          "opacity": 1,
          "paddingBottom": 9,
          "paddingLeft": 9,
          "paddingRight": 9,
          "paddingTop": 9,
        }
      }
      testID="logoutButton"
    >
      <Text>
        Icon
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#e5e5e5",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginTop": 32,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 0,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "borderColor": "#33333333",
              "borderRadius": 2,
              "borderStyle": "solid",
              "borderWidth": 1,
              "flexDirection": "row",
              "justifyContent": "space-between",
              "width": "90%",
            },
          ]
        }
      >
        <TextInput
          onChangeText={[Function]}
          onEndEditing={[Function]}
          placeholder="Procure um livro"
          style={
            Array [
              Object {
                "paddingBottom": 16,
                "paddingLeft": 8,
                "paddingRight": 46,
                "paddingTop": 16,
                "width": "100%",
              },
            ]
          }
          testID="searchInput"
          value=""
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "marginBottom": 16,
              "marginLeft": 16,
              "marginRight": 16,
              "marginTop": 16,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
            }
          }
          testID="searchInputButton"
        >
          <Text>
            Icon
          </Text>
        </View>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text>
          Icon
        </Text>
      </View>
    </View>
    <RCTScrollView
      ListFooterComponent={
        <BooksFooter
          disabled={false}
          onLoadMore={[Function]}
        />
      }
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      testID="bookList"
      viewabilityConfigCallbackPairs={Array []}
    >
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                },
              ]
            }
          >
            <View
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      isVisible={false}
      onBackdropPress={[Function]}
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "padding": 16,
        }
      }
      testID="filtersModal"
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 4,
              "paddingBottom": 24,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
              "width": "100%",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderColor": "#33333333",
                "borderRadius": 100,
                "borderStyle": "solid",
                "borderWidth": 1,
                "opacity": 1,
                "paddingBottom": 9,
                "paddingLeft": 9,
                "paddingRight": 9,
                "paddingTop": 9,
              }
            }
            testID="closeButton"
          >
            <Text>
              Icon
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "marginBottom": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 16,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "marginBottom": 4,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 4,
                },
              ]
            }
            testID="filtersChipsTitle"
          >
            Selecione a categoria
          </Text>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "marginBottom": 4,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 4,
                },
              ]
            }
            testID="filtersChipsOptions"
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderColor": "#333333",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "marginBottom": 0,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 0,
                  "opacity": 1,
                  "paddingBottom": 8,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 8,
                }
              }
              testID="filtersChipsOptionButtonCategory1"
            >
              <Text
                isSelected={false}
                style={
                  Array [
                    Object {
                      "color": "#333333",
                      "fontSize": 12,
                    },
                  ]
                }
                testID="filtersChipsOptionTextCategory1"
              >
                Design
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderColor": "#333333",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "marginBottom": 0,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 0,
                  "opacity": 1,
                  "paddingBottom": 8,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 8,
                }
              }
              testID="filtersChipsOptionButtonCategory2"
            >
              <Text
                isSelected={false}
                style={
                  Array [
                    Object {
                      "color": "#333333",
                      "fontSize": 12,
                    },
                  ]
                }
                testID="filtersChipsOptionTextCategory2"
              >
                UX Design
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderColor": "#333333",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "marginBottom": 0,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 0,
                  "opacity": 1,
                  "paddingBottom": 8,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 8,
                }
              }
              testID="filtersChipsOptionButtonCategory3"
            >
              <Text
                isSelected={false}
                style={
                  Array [
                    Object {
                      "color": "#333333",
                      "fontSize": 12,
                    },
                  ]
                }
                testID="filtersChipsOptionTextCategory3"
              >
                UI Design
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "marginBottom": 16,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 16,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontSize": 12,
                  "fontWeight": "bold",
                  "marginBottom": 4,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 4,
                },
              ]
            }
            testID="filtersChipsTitle"
          >
            Selecione o ano
          </Text>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "marginBottom": 4,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 4,
                },
              ]
            }
            testID="filtersChipsOptions"
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderColor": "#333333",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "marginBottom": 0,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 0,
                  "opacity": 1,
                  "paddingBottom": 8,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 8,
                }
              }
              testID="filtersChipsOptionButtonYear1"
            >
              <Text
                isSelected={false}
                style={
                  Array [
                    Object {
                      "color": "#333333",
                      "fontSize": 12,
                    },
                  ]
                }
                testID="filtersChipsOptionTextYear1"
              >
                1998
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderColor": "#333333",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "marginBottom": 0,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 0,
                  "opacity": 1,
                  "paddingBottom": 8,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 8,
                }
              }
              testID="filtersChipsOptionButtonYear2"
            >
              <Text
                isSelected={false}
                style={
                  Array [
                    Object {
                      "color": "#333333",
                      "fontSize": 12,
                    },
                  ]
                }
                testID="filtersChipsOptionTextYear2"
              >
                2000
              </Text>
            </View>
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ffffff",
                  "borderColor": "#333333",
                  "borderRadius": 20,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "marginBottom": 0,
                  "marginLeft": 4,
                  "marginRight": 4,
                  "marginTop": 0,
                  "opacity": 1,
                  "paddingBottom": 8,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 8,
                }
              }
              testID="filtersChipsOptionButtonYear3"
            >
              <Text
                isSelected={false}
                style={
                  Array [
                    Object {
                      "color": "#333333",
                      "fontSize": 12,
                    },
                  ]
                }
                testID="filtersChipsOptionTextYear3"
              >
                2002
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "borderColor": "#B22E6F",
                "borderRadius": 20,
                "borderStyle": "solid",
                "borderWidth": 1,
                "justifyContent": "center",
                "marginTop": 32,
                "opacity": 1,
                "paddingBottom": 8,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 8,
              }
            }
            testID="filterButton"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#B22E6F",
                    "fontSize": 16,
                  },
                ]
              }
            >
              Filtrar
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
